### PR TITLE
Add the form to add items to the list

### LINKED ISSUE #4: 4. As a user, I want to add a new item to my shopping list so I can start recording purchases.
A shopping list item consists of the following data points:

* Name of item
* How soon are you likely to buy it again?
  * Soon (in the next 7 days)
  * Kind of soon (in the next 14 days)
  * Not soon (in the next 30 days)
* Last purchased date

AC:

* User is presented with a form that allows them to enter the name of the item and select how soon they anticipate needing to buy it again (Soon, Kind of Soon, Not Soon)
* When the user submits the form, the item is saved to the database, associated with the user’s token
* Along with the item name, an integer corresponding to the estimated number of days until next purchase is saved: `7` for “Soon”, `14` for “Kind of Soon”, and `30` for “Not Soon”
* The last purchased date should be set to `null` initially (i.e. you can create an item without purchasing it)
* Item names should be displayed as a simple, unordered list on the list view after they’re added
Notes:
* The add item form has the potential to be a lot of scope. The wireframe https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/add-item.png has several notes related to accessibility and UX.
* Eventually, the list will look like the following wireframe: https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/list-populated.png
* It's good for your team to look at the wireframe now, but be sure not to try to implement all of the visible functionality at once. There will be future stories to add all of those features.
* For this story, it's sufficient just to get a simple list of the items displaying.

### PR DESCRIPTION
**Pull request by redapy**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description
- This PR adds a `<AddItemForm/>` components inside the `<AdItem />`.
### `<AddItemForm/>`
- Created from that has controlled inputs, that fire a `handleSubmit` function on submit.
- Used the `addItem` function (created issue 1)  inside handleSubmit to send the user inputs with a token (for now we are using a dummy token)
### Listen to changes in the list view (fetch data from firestore)
- Moved `onSnapshot` from `App.js` to `List.js`.
- Provide  `onSnapshot` with a query parameter that will fetch only the data that have the user token.

 
## Related Issue
 Close #4  

## Acceptance Criteria
- User is presented with a form that allows them to enter the name of the item and select how soon they anticipate needing to buy it again (Soon, Kind of Soon, Not Soon)
- When the user submits the form, the item is saved to the database, associated with the user’s token
- Along with the item name, an integer corresponding to the estimated number of days until next purchase is saved: 7 for “Soon”, 14 for “Kind of Soon”, and 30 for “Not Soon”
- The last purchased date should be set to null initially (i.e. you can create an item without purchasing it)
- Item names should be displayed as a simple, unordered list on the list view after they’re added

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Testing Steps / QA Criteria
- From your terminal, pull down this branch with `git pull origin rb-to-form-to-add-new-item` and check that branch out with `git checkout rb-to-form-to-add-new-item`.
- Click on the "add an item" link.
-Type the item name and chose how soon you will need to buy it, then click the "add item" button
- Go to the list view, and you will find the item name of the item you added.
- Go to firestore, and you will notice that there are other items there but they are not shown. that because they don't have the token

### CODE REVIEW

**Comment by redapy** [comment]
I just created the `<AddItemForm />` where we can put the related code to this PR.

**Comment by hellodeborahuk** [comment]
Great work @redapy and @OloyedeTobi, this worked for me. Should it go to the list view once you've submitted the item? I couldn't tell either way from the AC.

**Comment by JimeBlue** [comment]
Great implementation @redapy and @OloyedeTobi, very clear and up to the point! It´s working for me as well. In my view, the AC does not require to redirect to the list view after an itme is submitted, so the issue is complete as it is.

**Comment by redapy** [comment]
@hellodeborahuk as @JimeBlue  said I don't think it should be redirected to the list view, at least not for this issue. I'll add a small function that clears the input after an item is added.

File: src/components/AddItemForm.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import { useState } from 'react';
+import { db } from '../lib/firebase';
+import { addDoc, collection } from 'firebase/firestore';
+
+const AddItemForm = () => {
+  const [timeframe, setTimframe] = useState('7');
```
Comment (faysvas) [inline src/components/AddItemForm.js:7]: Minor typo here `setTimFrame` :slightly_smiling_face:

File: src/components/List.js [hunk 2]
```diff
@@ -1,5 +1,27 @@
...
+import { db } from '../lib/firebase';
+import { onSnapshot, collection, query, where } from 'firebase/firestore';
+
 const List = () => {
-  return <div>List</div>;
+  const [datas, setData] = useState([]);
+  useEffect(() => {
+    const ListRef = collection(db, 'List1');
+    const q = query(ListRef, where('token', '==', 'reda and tobi'));
+    const unsb = onSnapshot(q, ListRef, (snapshot) => {
```
Comment (faysvas) [inline src/components/List.js]: Maybe these two variables `q` and `unsb` can have more descriptive names so they're easier to read and understand :slightly_smiling_face: ?

File: src/components/AddItemForm.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        <label htmlFor="kind-of-soon">kind of soon</label>
+        <input
+          type="radio"
+          id="not-soon"
+          value="30"
+          checked={timeframe === '30'}
+          onChange={handleSelect}
+          name="timeframe"
+        />
+        <label htmlFor="not-soon">not-Soon</label>
```
Comment (faysvas) [inline src/components/AddItemForm.js:81]: Not very important, but you can write "Not Soon" here to more closely follow the wireframe :slightly_smiling_face: https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/add-item.png

File: src/components/AddItemForm.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        <label htmlFor="soon">Soon</label>
+        <input
+          type="radio"
+          id="kind-of-soon"
+          value="14"
+          checked={timeframe === '14'}
+          onChange={handleSelect}
+          name="timeframe"
+        />
+        <label htmlFor="kind-of-soon">kind of soon</label>
```
Comment (faysvas) [inline src/components/AddItemForm.js:72]: Not very important, but you can write "Kind of Soon" here to more closely follow the wireframe :slightly_smiling_face: https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/add-item.png

**Pull request merged**